src/tuner/nccl_ofi_model.c

```diff
@@ -74,180 +66,261 @@ static const float nccl_base_lat[NCCL_NUM_ALGORITHMS][NCCL_NUM_PROTOCOLS] = {
+	const double base_latency = nccl_base_lat[NCCL_ALGO_RING][proto];
 	const double p2p_latency = nccl_nvlink_lat[NCCL_ALGO_RING][proto];
+	const double internodeSteps = 2.0 * (dims->num_nodes);
+	const double intranodeSteps = 2.0 * (dims->num_ranks - 1) - internodeSteps;
 
-	const double logranks = log(dims->num_ranks) / log(2);
-	const double logbytes = log(nBytes) / log(2);
-	const double lograt = (logbytes - logranks);
-	if (floor(lograt) < 23) {
-		/* Hack until we understand this better. */
-		return INFINITY;
-	}
+	const double internodeLatency =
+		internodeSteps *
+		(base_latency + NCCL_OFI_TUNER_NETWORK_LATENCY +
+		 ((proto == NCCL_PROTO_SIMPLE) ? NCCL_OFI_TUNER_COMPLETION_OVERHEAD : 0));
+	const double intranodeLatency = (base_latency + p2p_latency) * intranodeSteps;
+	const double latencyCost = intranodeLatency + internodeLatency;
 
-	const double steps_per_rotation = dims->num_ranks;
-	const double num_rotations = 2;
-	const double num_steps = (num_rotations * steps_per_rotation) - 1;
-	const double num_steps_internode = (num_rotations * dims->num_nodes);
-	const double num_steps_intranode = num_steps - num_steps_internode;
-
-	double per_proto_bandwidth_overhead;
-	double net_latency = NCCL_OFI_TUNER_NETWORK_LATENCY;
-	switch (proto) {
-	case NCCL_PROTO_LL: {
-		per_proto_bandwidth_overhead = NCCL_OFI_TUNER_LL_PROTO_BW_FACTOR;
-		break;
-	}
-	case NCCL_PROTO_LL128: {
-		per_proto_bandwidth_overhead = NCCL_OFI_TUNER_LL128_PROTO_BW_FACTOR;
-		break;
+	const double internodeBw = 1.0 * NCCL_OFI_TUNER_INTERNODE_BW * chan / 2 / dims->num_ranks;
+	const double intranodeBw = 1.0 * NCCL_OFI_TUNER_INTRANODE_BW * dims->num_ranks;
 
+	double effective_message_size = nBytes;
+	if (proto == NCCL_PROTO_LL) {
+		effective_message_size *= 2;
+	} else if (proto == NCCL_PROTO_LL) {
+		effective_message_size += (nBytes / 15);
 	}
-	case NCCL_PROTO_SIMPLE: {
-		per_proto_bandwidth_overhead = NCCL_OFI_TUNER_SIMPLE_PROTO_BW_FACTOR;
-		net_latency += NCCL_OFI_TUNER_COMPLETION_OVERHEAD;
-		break;
-	}
-	default: {
-		return INFINITY;
-	}
-	}
-
-
-	const double intranode_latency = p2p_latency * num_steps_intranode;
-	const double intranode_bw = (NCCL_OFI_TUNER_INTRANODE_BW);
-	const double intranode_bytes = nBytes * (num_steps_intranode / num_steps);
-	const double intranode_cost = intranode_latency + (intranode_bytes / intranode_bw);
-
-	const double internode_latency = net_latency * num_steps_internode;
-	const double internode_bw = (NCCL_OFI_TUNER_INTERNODE_BW) * per_proto_bandwidth_overhead;
-	const double internode_bytes = nBytes * (num_steps_internode / num_steps);
-	const double internode_cost = internode_latency + (internode_bytes / internode_bw);
 
-	/* XXX: Should be scaled further here, not done for the sake of ensuring
-	 * this is cheaper than NVLSTree at dims where we did not explicitly opt
-	 * out of choosing ring above */
-	return ((base_latency * num_steps) + intranode_cost + internode_cost);
+	double totalCost =
```

Comment:
why x2 to effective_message_size for totalCost calcuation?
The same x2 on line 107 and 108 seem redundant.
Can we remove x2 for all of these internode bandwidth cost (line 105), intranode bandwidth cost (line 107) and latency cost (line 108)?

Reply:
additional comment is made for latencyCost in line 108

Reply:
Resolving as outdated -- please reopen if you still feel this is broken.